modules/core/src/lib/timeline.js

```diff
@@ -0,0 +1,94 @@
+const WRAP_LOOP = 0;
+const WRAP_CLAMP = 1;
+const WRAP_MAP = {
+  loop: WRAP_LOOP,
+  clamp: WRAP_CLAMP
+};
+
+export class Timeline {
+  constructor() {
+    this.time = 0;
+    this.duration = Number.POSITIVE_INFINITY;
+    this.wrapMode = WRAP_LOOP;
+    this.channels = [];
+    this.rate = 1;
+    this.playing = false;
+    this.lastEngineTime = -1;
+  }
+
+  addChannel(props) {
+    const {duration = Number.POSITIVE_INFINITY, wrapMode = 'loop', rate = 1} = props;
+
+    const handle = this.channels.length;
+    const channel = {
+      time: 0,
```

Comment:
Can channels have delay/offset?

Reply:
Definitely doable. Was planning for it, but thought I'd simplify for a first iteration. Do you want it in now?